Port flags to Rust (#13)

dmitryrec opened the pull request:
Closes #13

dmitryrec committed:
Init PR

ales-tsurko commented:
Для удобства:

* [X] Plugin flags
* [X] Processing Mode Flags
* [x] ProcessParam Flags
* [x] Parameter Popup Menu Item Flags
* [x] Sample Loading Flags
* [x] TNotesParams Flags
* [X] Parameter Flags
* [x] GetInBuffer / GetOutBuffer flags

dmitryrec committed:
Add process param flags

dmitryrec committed:
Update lib.rs file
- add  Parameter popup menu item flags
- add Sample loading flags
- add TNotes params flags

dmitryrec committed:
Port more types
- add GetInBuffer and GetOutBuffer flags

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+        const SHOW_HINT = 4;
+        /// Update the parameter control (wheel, slider, ...)
+        const UPDATE_CONTROL = 16;
+        /// A value between 0 and 65536 has to be translated to the range of the parameter control.
```
Добавь пустую строку после этой (в смысле доккоммент пустой строки), потому что тут перенос. В markdown строки объединяются, если между ними нет пустой.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+        /// Update the parameter control (wheel, slider, ...)
+        const UPDATE_CONTROL = 16;
+        /// A value between 0 and 65536 has to be translated to the range of the parameter control.
+        /// Note that you should also return the translated value, even if REC_GetValue isn't included.
```
Доку тоже портировать надо. Не `REC_GetValue`, а `ProcessParamFlags::GET_VALUE`. И еще должна быть ссылка на него: `[ProcessParamFlags::GET_VALUE](struct.ProcessParamFlags.html#associatedconstant.GET_VALUE)`

В дальнейшем, чтобы разобраться, как ссылки генерятся для доков в Rust, скомпиль уже существующую доку (`cargo doc --no-deps`), главная будет `target/doc/fpsdk/index.html`. Там можешь просто по ссылкам покликать и посмотреть, что будет в адресной строке.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+
+bitflags! {
+    /// Parameter popup menu item flags
+    pub struct ParameterPopupMenuItemFlags: isize {
```
Слишком длинное имя. Удобно такое набирать?) Переименуй в `ParamMenuItemFlags`.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+        /// Force the sample to be reloaded, even if the filename is the same.
+        /// This is handy in case you modified the sample, for example
+        const FORCE_RELOAD = 2;
+        /// Don't load the sample, instead get its filename & make sure that the format is correct
```
То же самое с разрывом строки.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+        /// This is handy in case you modified the sample, for example
+        const FORCE_RELOAD = 2;
+        /// Don't load the sample, instead get its filename & make sure that the format is correct
+        /// (useful after FPD_ChanSampleChanged)
```
То же самое с портом типов в доках. Этот портирован в `host::HostMessage::ChanSampleChanged`. Поменяй и сделай на него ссылку, пожалуйста.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+
+bitflags! {
+    /// Sample loading flags
+    pub struct SampleLoadingFlags: isize {
```
Давай переименуем в `SampleLoadFlags`.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+
+bitflags! {
+    /// TNotes parameters flags
+    pub struct TNotesParamsFlags: isize {
```
Здесь `T` не нужно перед именем. Это в C/C++ есть смысл, где T и P разделены как конкретный тип и поинтер на него (алиас). Имена тоже нужно портировать в идиоматичный Rust. Проверяй соответствующий тип в C/C++ предварительно, чтобы понять зачем `T`, например.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+}
+
+bitflags! {
+    /// TNotes parameters flags
```
В доке не забудь `T` про которое ниже написал.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+bitflags! {
+    /// GetInBuffer flags
+    pub struct GetInBufferFlags: isize {
+        /// before adding to the buffer
```
Сделай с заглавной буквы. Везде ж с заглавной.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+    /// GetInBuffer flags
+    pub struct GetInBufferFlags: isize {
+        /// before adding to the buffer
+        const IO_LOCK = 0;
```
Давай эти `IO_` флаги (включая нижний для out buffer) оставим константами. Определи их просто как константы, без `bitflags!`.

ales-tsurko reviewed (changes requested):
По докам я мог пропустить что-то или не все отписать. Проверь внимательнее, чтобы это все хорошо выглядело и понятно было.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,84 @@ bitflags! {
+    /// GetOutBuffer flag
+    pub struct GetOutBuffer: isize {
+        /// tells if the buffer is filled
+        const IO_FILLED = 0;
```
Здесь неправильное значение.

dmitryrec committed:
Fix structure
- add links
- add empty comment lines
- make constants

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,81 @@ bitflags! {
+        /// A value between 0 and 65536 has to be translated to the range of the parameter control.
+        /// 
+        /// Note that you should also return the translated value, even if 
+        /// 
```
Здесь разрыв строки явно не нужен) Оно же inline, это же ссылка внутри текста.

Проверяй доки вручную, смотри, что получается.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,81 @@ bitflags! {
+
+    /// GetInBuffer flags
+        /// Before adding to the buffer
+        const IO_LOCK: i32 = 0;
```
Ну, пусть будут `i32`, хорошо. Но константы не по [гайду](https://github.com/tonikasoft/tonika-docs/blob/master/CODE_STYLE.md#priority-of-blocks-in-document-from-high-to-low). И зачем ты отступы добавил? Плюс, опять же, посмотри, что у тебя с этим в доках получится. Надо каждую константу из этих отдельно документировать. Например для `IO_LOCK`:

`GetInBuffer` flag, which is added before adding to the buffer.

По остальным посмотри сам, подумай, как лучше.

ales-tsurko commented on src/lib.rs:
```diff
@@ -298,6 +298,81 @@ bitflags! {
+        const FORCE_RELOAD = 2;
+        /// Don't load the sample, instead get its filename & make sure that the format is correct
+        /// 
+        /// (useful after [host::HostMessage::ChanSampleChanged](struct.SampleLoadFlags.html#associatedconstant.GET_NAME))
```
Ну и куда эта ссылка поведет?)

dmitryrec committed:
Fix file structure and doc

dmitryrec committed:
Fix constant
- make public
- paste in correct place

dmitryrec committed:
Merge branch 'master' of https://github.com/tonikasoft/fpsdk into 13-port-flags-rust

ales-tsurko committed:
Re-format

ales-tsurko reviewed (approved):
🎆
